src/jitlayers.cpp

```diff
@@ -1223,6 +869,73 @@ std::unique_ptr<jitlink::JITLinkMemoryManager> createJITLinkMemoryManager() JL_N
 #endif
 }
 
+class JLMaterializationUnit : public orc::MaterializationUnit {
+public:
+    static JLMaterializationUnit Create(JuliaOJIT &JIT, ObjectLinkingLayer &OL,
+                                        std::unique_ptr<jl_linker_info_t> Info,
+                                        std::unique_ptr<MemoryBuffer> Obj) JL_NOTSAFEPOINT
+    {
```

Comment:
Nice! I have been wanting this for a long time!

Would it make sense to have a C-API for creating these? So that LLVM.jl could create them?

Reply:
Yeah I have been wanting to try and ORC based setup for GPUCompiler